cranelift/fuzzgen/src/function_generator.rs

```diff
@@ -1614,7 +1614,15 @@ where
             }
             BlockTerminator::BrTable(default, targets) => {
                 // Create jump tables on demand
-                let jt = builder.create_jump_table(JumpTableData::new(default, &targets));
+                let mut jt = Vec::with_capacity(targets.len());
+                for block in targets {
+                    let args = self.generate_values_for_block(builder, block)?;
+                    jt.push(builder.func.dfg.block_call(block, &args))
+                }
+
+                let args = self.generate_values_for_block(builder, default)?;
+                let jt_data = JumpTableData::new(builder.func.dfg.block_call(default, &args), &jt);
+                let jt = builder.create_jump_table(jt_data);
```

Comment:
This isn't sufficient to generate br_table with block params. We also need to update the match on `BlockTerminatorKind::BrTable` elsewhere to use `generate_target_block` instead of selecting from `forward_blocks_without_params`. Also that should be the last use of the latter as well as a variety of other machinery which can all be removed now. Hooray!

Reply:
Alongside what @jameysharp is saying above, we can now select `Switch` and `BrTable` more often, since they no longer require a *special* block without params.

Could you also update this: https://github.com/bytecodealliance/wasmtime/blob/e10094dcd6d0354628255a6f2e69c1e4c327d6e7/cranelift/fuzzgen/src/function_generator.rs#L1806-L1815  So that we allow `BrTable` and `Switch` alongside `Jump` and `Br`?


The only other places where we use `forward_blocks_without_params`, is when selecting blocks for `Switch` targets, and those can also now carry params, so I agree, lets remove it!

Reply:
I believe I've made the changes you both requested, could you take a quick look and make sure?

Reply:
LGTM! 👍